crates/holochain/src/sweettest/sweet_conductor_config_rendezvous.rs

```diff
@@ -1,12 +1,31 @@
 use std::sync::{Arc, Mutex};
 
+use once_cell::sync::OnceCell;
+
+/// Rendezvous for local testing.
+pub static RENDEZVOUS: OnceCell<std::sync::Weak<dyn SweetRendezvous>> = OnceCell::new();
+
+/// Get a shared local rendezvous for testing.
+pub async fn shared_rendezvous() -> Arc<dyn SweetRendezvous> {
+    if let Some(shared) = RENDEZVOUS.get() {
+        shared.upgrade().unwrap()
+    } else {
+        let vous = SweetLocalRendezvous::new().await;
+        let shared = RENDEZVOUS
+            .get_or_init(|| Arc::downgrade(&vous))
+            .upgrade()
+            .unwrap();
+        drop(vous);
+        shared
+    }
+}
```

Comment:
I think there is a race condition in here with the upgrades if the last strong is dropped after the get but before the upgrade.

Also, the only reason I can see to try to drop the rendezvous after a tests is to try to avoid the bootstrap:true versions from overlapping, but this doesn't accomplish that. Because if a test binary is running two tests in parallel and they are both bootstrap:true, they may end up sharing a bootstrap server with this implementation.

I think this would be simpler and less error-prone if it used a `OnceCell<futures::future::Shared<Arc<dyn SweedRendezvous>>>` that was only ever initialized once, and then just awaited on for every call of this function.

Reply:
@neonphog I was trying to respect the Drop impl, but by your logic, I think the Drop is also ineffective. Should that be removed as well?

https://github.com/holochain/holochain/blob/4b82f959837c64e1ebcf84552ea3ab4b847f84de/crates/holochain/src/sweettest/sweet_conductor_config_rendezvous.rs#L48-L59

Reply:
@maackle - No, the drop is fine. In a single test, you can create however many rendezvous you want, use them, and then they'll drop when the test goes out of scope.

The problem is introduced by your "shared" rendezvous. Because if one test is done with it, but another test grabs a handle at just the wrong moment, race conditions ensue. To me, it seems better in the race condition case, to just never drop it.

Reply:
@neonphog I see, I forgot that tests are run in parallel in the same process, was wishfully thinking that each would have its own process.

Well then this pattern is probably no good, since agents across tests will connect to the same bootstrap.

What you're suggesting would cause not simply some tests in the binary, but *all* tests in the binary to share the same bootstrap server. And that still creates a race condition, because then the order in which tests run matters, since that will determine which unavailable agents are present in the shared bootstrap server at the time.

I was trying to create the least possible amount of changes needed to port things over to tx5. I also wrote this before your tx5 mem server was available. Perhaps with that, I wouldn't have needed to lean so heavily on rendezvous, which was the only way to get a valid test network set up. But alas, I probably won't have time to revert that on my last day. If the shared rendezvous becomes a problem, I'd suggest porting these tests back over to mem and manually exchanging agent info where appropriate.

Reply:
@maackle - What about a hack to get this across the finish line where you put a hashmap in there with the current [Thread::id](https://doc.rust-lang.org/std/thread/struct.Thread.html#method.id) as the key?

Reply:
Don't think that works, because two conductors on different threads in the same test could legitimately need to share a rendezvous

Reply:
That's up to the setup of the test, right? You should be able to pass in the same rendezvous to the setup, or if you want to use this "shared" method, create the multiple conductors in the main thread of the test before passing them off into separate tasks. no?

Reply:
I guess it would probably just be cleaner to do away with this shared concept. If you want me to take over doing that work, I'm up for it.

Reply:
I only did it this way to minimize lines changed, for the many tests that were still using a tx2 mem server. Most of the shared_rendezvous tests could use a tx5 mem server now, and the ones that can't could configure the rendezvous explicitly.

Yes, please take that over since I won't be able to finish that today.